### PR TITLE
feat(scheduler): pipe Katana output to job logs

### PR DESCRIPTION
**Pull request by boxbeam**
image

### CODE REVIEW

File: crates/tabby-scheduler/src/crawl/mod.rs [hunk 1]
```diff
@@ -35,6 +41,12 @@ async fn crawl_url(start_url: &str) -> anyhow::Result<impl Stream<Item = KatanaR
         }
     });
 
+    tokio::spawn(async move {
+        while let Ok(Some(line)) = stderr.next_line().await {
+            on_stderr_line(line);
```
Comment (wsxiaoys) [inline crates/tabby-scheduler/src/crawl/mod.rs:46]: just invoke logkit::info here?

File: ee/tabby-webserver/src/service/background_job/web_crawler.rs [hunk 2]
```diff
@@ -19,9 +19,17 @@ impl WebCrawlerJob {
...
 
     pub async fn run(self, embedding: Arc<dyn Embedding>) -> tabby_schema::Result<()> {
-        tabby_scheduler::crawl_index_docs(&self.source_id, &self.url, embedding, move |url| {
-            logkit::info!("Fetching {}", url);
-        })
+        tabby_scheduler::crawl_index_docs(
+            &self.source_id,
+            &self.url,
+            embedding,
+            move |url| {
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/background_job/web_crawler.rs:26]: ditto - you can refactor this out and just invoke logkit::info directly, without this callback.

File: crates/tabby-scheduler/src/crawl/mod.rs (hunk 1 above)
Comment (boxbeam) [reply crates/tabby-scheduler/src/crawl/mod.rs:46]: Got it, wasn't aware this would work across both crates

Comment (boxbeam) [reply crates/tabby-scheduler/src/crawl/mod.rs]: Can't we do the same for `on_process_url` too then?

Comment (wsxiaoys) [reply crates/tabby-scheduler/src/crawl/mod.rs]: I commented on this in last round, guess it's missed?
https://github.com/TabbyML/tabby/pull/2519#discussion_r1657991256

**Pull request merged**